Framework/Kernel/src/DateAndTimeHelpers.cpp

```diff
@@ -0,0 +1,86 @@
+#include "MantidKernel/DateAndTimeHelpers.h"
+#include "MantidKernel/Logger.h"
+#include <Poco/DateTime.h>
+#include <Poco/DateTimeFormat.h>
+#include <Poco/DateTimeParser.h>
+
+namespace {
+std::tuple<bool, size_t, std::string> isARGUSDateTime(const std::string &date) {
+  // Some ARGUS files have an invalid date with a space instead of zero.
+  // To enable such files to be loaded we correct the date and issue a warning
+  // (ticket #4017).
+  // just take the date not the time or any date-time separator
+  std::string strippedDate = date.substr(0, 10);
+  const size_t nSpace = strippedDate.find(' ');
+  return std::make_tuple(nSpace != std::string::npos, nSpace, strippedDate);
+}
+} // namespace
+
+namespace Mantid {
+namespace Kernel {
+namespace DateAndTimeHelpers {
+// Initialize the logger
+Logger g_log("DateAndTime");
+
+/** Creates a DateAndTime object from a date string even if the string does not
+ *exactly conform to ISO8601 (ARGUS File)
+ *@param date Date used to create DateAndTime object. May be sanitized first.
+ *
+ */
+DateAndTime createFromSanitizedISO8601(const std::string &date) {
+  return DateAndTime(verifyAndSanitizeISO8601(date));
+}
+
+/** Check if a string is iso8601 format.
+ *
+ * @param date :: string to check
+ * @return true if the string conforms to ISO 860I, false otherwise.
+ */
+bool stringIsISO8601(const std::string &date) {
+  Poco::DateTime dt;
+  int tz_diff;
+  return Poco::DateTimeParser::tryParse(Poco::DateTimeFormat::ISO8601_FORMAT,
+                                        date, dt, tz_diff);
+}
+
+/** Verifies whether or not a string conforms to ISO8601. Corrects the string
+ *if it does not and is of the ARGUS file date/time format
+ *
```

Comment:
For future readers, i think it would be useful to have something in the comments here, or for `isArgusDateTime` which gives and example of what that date-time string is like. and what the corresponding sanitized string ends up like. I wouldn't object if this was done as a separate PR, as I know that this PR is holding up others.

Reply:
This will be added in this PR as changes are made there for DateAndTime anyway https://github.com/mantidproject/mantid/pull/20481